### ISSUE TITLE
capistrano-magento2 - di:compile errors not reported

### ISSUE THREAD

**Issue by davidalger** [davidalger/capistrano-magento2#25]
I ran a deployment to prod (bundle exec cap prod deploy) and it did not report any errors during the di:compile step.

00:46 magento:setup:static-content:deploy
      01 php -f bin/magento -- setup:static-content:deploy | stdbuf -o0 tr -d .
    ✔ 01 www-prod@74.205.39.4 62.177s
01:49 magento:setup:di:compile
      01 php -f bin/magento -- setup:di:compile-multi-tenant
    ✔ 01 www-prod@74.205.39.4 38.825s
02:28 magento:maintenance:enable
      01 php -f bin/magento -- maintenance:enable
    ✔ 01 www-prod@74.205.39.4 0.706s

I only attempted this because I saw the same error when trying to recreate an issue on my local dev machine by running my dev environment in production mode which caused me to run the di:compile step by hand on my dev environment.

When I ran di:compile manually on the production environment it produced an error at the end:

```
$ bin/magento setup:di:compile-multi-tenant
Errors during compilation:
    ClientNamespace\GetRewardsPoints\Block\Product\Rewards
        Incorrect dependency in class ClientNamespace\GetRewardsPoints\Block\Product\Rewards in /var/www/prod/releases/20160831142430/app/code/ClientNamespace/GetRewardsPoints/Block/Product/Rewards.php
\Magento\Store\Model\StoreManagerInterface already exists in context object
Total Errors Count: 1
```

report originally from @mttjohnson

**Comment by davidalger** [comment]
Unfortunately the Magento CLI does not report errors by returning a proper exit code in 2.0 ( I think 2.1 was supposed to fix this ). So the static content generator output is being checked for the presence of a success message and throws an error if it does not exist.

Note that the output of the command is recorded in the capistrano log file where output has been limited to stderr output (typical in our deployments) or is merely displayed, but does not stop the process (which it should).

Would one of you guys be able to send me a commit hash (privately) which will produce this error when deployed? That will give me what I need to attempt deployment to a local vm and implement a similar output check on the command.

Thanks for reporting this!

## 

David Alger
Director of Technology, Classy Llama

**Comment by davidalger** [comment]
The fix is in place. The following is what should happen where the setup:di:compile-multi-tenant fails.

```
10:19 magento:setup:di:compile
      01 php -f bin/magento -- setup:di:compile-multi-tenant
    ✔ 01 vagrant@dev-web70 162.711s
(Backtrace restricted to imported tasks)
cap aborted!
Exception: setup:di:compile-multi-tenant command execution failed

Tasks: TOP => magento:setup:di:compile
(See full trace by running task with --trace)
The deploy has failed with an error: setup:di:compile-multi-tenant command execution failed

** DEPLOY FAILED
** Refer to log/capistrano.log for details. Here are the last 20 lines:

 DEBUG [ab57dce6]       Xtento\ProductExport\Block\Adminhtml\Log\Interceptor
 DEBUG [ab57dce6]       Xtento\ProductExport\Block\Adminhtml\Destination\Interceptor
 DEBUG [ab57dce6]       Xtento\ProductExport\Block\Adminhtml\Destination\Edit\Interceptor
 DEBUG [ab57dce6]       Xtento\ProductExport\Block\Adminhtml\Profile\Interceptor
 DEBUG [ab57dce6]       Xtento\ProductExport\Block\Adminhtml\Profile\Edit\Interceptor
 DEBUG [ab57dce6]       Xtento\ProductExport\Block\Adminhtml\History\Interceptor
 DEBUG [ab57dce6]       Xtento\ProductExport\Model\System\Config\Backend\Server\Interceptor
 DEBUG [ab57dce6]       Magento\Framework\Controller\Index\Index\Interceptor
 DEBUG [ab57dce6]       Magento\Framework\Controller\Noroute\Index\Interceptor
 DEBUG [ab57dce6]       Magento\Framework\Controller\Result\Raw\Interceptor
 DEBUG [ab57dce6]       Magento\Framework\Controller\Result\Json\Interceptor
 DEBUG [ab57dce6]       Magento\Framework\App\Action\Redirect\Interceptor
 DEBUG [ab57dce6]       Magento\Framework\App\Action\Forward\Interceptor
 DEBUG [ab57dce6]       Magento\Framework\Mail\Transport\Interceptor
 DEBUG [ab57dce6]   Errors during compilation:
 DEBUG [ab57dce6]       ClientNamespace\GetRewardsPoints\Block\Product\Rewards
 DEBUG [ab57dce6]           Incorrect dependency in class ClientNamespace\GetRewardsPoints\Block\Product\Rewards in /home/vagrant/sites/wll-test/releases/20160908174456/app/code/ClientNamespace/GetRewardsPoints/Block/Product/Rewards.php
\Magento\Store\Model\StoreManagerInterface already exists in context object
 DEBUG [ab57dce6]   Total Errors Count: 1
  INFO [ab57dce6] Finished in 162.711 seconds with exit status 0 (successful).
```

**Comment by davidalger** [comment]
Fixed in 0.5.0
